File: backend/src/main/java/org/akashihi/mdg/dao/AccountTypeConverter.java
package org.akashihi.mdg.dao;

import org.akashihi.mdg.entity.AccountType;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
public class AccountTypeConverter implements AttributeConverter<AccountType, String> {
    @Override
    public String convertToDatabaseColumn(AccountType accountType) {
        return accountType.toDbValue();
    }

    @Override
    public AccountType convertToEntityAttribute(String s) {
        return AccountType.from(s);
    }
}

File: backend/src/main/kotlin/org/akashihi/mdg/dao/AccountTypeConverter.kt
package org.akashihi.mdg.dao

import org.akashihi.mdg.entity.AccountType
import org.akashihi.mdg.entity.AccountType.Companion.from
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter
class AccountTypeConverter : AttributeConverter<AccountType, String> {
    override fun convertToDatabaseColumn(accountType: AccountType): String {
        return accountType.toDbValue()
    }

    override fun convertToEntityAttribute(s: String): AccountType {
        return from(s)
    }
}
